x-pack/plugins/stack_connectors/server/connector_types/resilient/resilient.ts

```diff
@@ -0,0 +1,363 @@
+  ExecutorSubActionGetSeverityParamsSchema,
+  ExecutorSubActionHandshakeParamsSchema,
+  GetCommonFieldsResponseSchema,
+  GetIncidentTypesResponseSchema,
+  GetSeverityResponseSchema,
+} from './schema';
+import { formatUpdateRequest } from './utils';
+
+const VIEW_INCIDENT_URL = `#incidents`;
+
+export class ResilientConnector extends CaseConnector<ResilientConfig, ResilientSecrets> {
+  private urls: {
+    incidentTypes: string;
+    incident: string;
+    comment: string;
+    severity: string;
+  };
+
+  constructor(params: ServiceParams<ResilientConfig, ResilientSecrets>) {
+    super(params);
+
+    this.urls = {
+      incidentTypes: `${this.getIncidentFieldsUrl()}/incident_type_ids`,
+      incident: `${this.getOrgUrl()}/incidents`,
+      comment: `${this.getOrgUrl()}/incidents/{inc_id}/comments`,
+      severity: `${this.getIncidentFieldsUrl()}/severity_code`,
+    };
+
+    this.registerSubActions();
+  }
+
+  protected getResponseErrorMessage(error: AxiosError) {
+    if (!error.response?.status) {
+      return i18n.UNKNOWN_API_ERROR;
+    }
+    if (error.response.status === 401) {
+      return i18n.UNAUTHORIZED_API_ERROR;
+    }
+    return `API Error: ${error.response?.statusText}`;
+  }
+
+  private registerSubActions() {
+    this.registerSubAction({
+      name: SUB_ACTION.INCIDENT_TYPES,
+      method: 'getIncidentTypes',
+      schema: ExecutorSubActionGetIncidentTypesParamsSchema,
+    });
+
+    this.registerSubAction({
+      name: SUB_ACTION.SEVERITY,
+      method: 'getSeverity',
+      schema: ExecutorSubActionGetSeverityParamsSchema,
+    });
+
+    this.registerSubAction({
+      name: SUB_ACTION.FIELDS,
+      method: 'getFields',
+      schema: ExecutorSubActionCommonFieldsParamsSchema,
+    });
+
+    this.registerSubAction({
+      name: SUB_ACTION.INCIDENT,
+      method: 'getIncident',
+      schema: ExecutorSubActionGetIncidentParamsSchema,
+    });
+
+    this.registerSubAction({
+      name: SUB_ACTION.HANDSHAKE,
+      method: 'handshake',
+      schema: ExecutorSubActionHandshakeParamsSchema,
+    });
+  }
+
+  private getAuthHeaders() {
+    const token = Buffer.from(
+      this.secrets.apiKeyId + ':' + this.secrets.apiKeySecret,
+      'utf8'
+    ).toString('base64');
+
+    return { Authorization: `Basic ${token}` };
+  }
+
+  private getOrgUrl() {
+    const { apiUrl: url, orgId } = this.config;
+
+    return `${url}/rest/orgs/${orgId}`;
+  }
+
+  private getIncidentFieldsUrl = () => `${this.getOrgUrl()}/types/incident/fields`;
+
+  private getIncidentViewURL(key: string) {
+    const url = this.config.apiUrl;
+    const urlWithoutTrailingSlash = url.endsWith('/') ? url.slice(0, -1) : url;
+
+    return `${urlWithoutTrailingSlash}/${VIEW_INCIDENT_URL}/${key}`;
+  }
+
+  public async createIncident({
+    incident,
+  }: Record<string, unknown>): Promise<ExternalServiceIncidentResponse> {
+    try {
+      const incidentData = incident as Incident;
+      if (!incidentData?.name) {
```

Comment:
We need it because `createIncident` is called from psuToService subAction. `x-pack/plugins/actions/server/sub_action_framework/case.ts` 
Whose schema is
` incident: schema.object(
          { externalId: schema.nullable(schema.string()) },
          { unknowns: 'allow' }
        ),` 
It doesn't check for name while creating an incident.

Reply:
😨 this is a bug on the framework level! We need to fix it somehow. Let's talk offline about it.

Reply:
Should we then call `schema.validate` here to ensure we check everything and not just the `name`?